src/mrsimulator/spin_system/isotope.py

```diff
@@ -176,6 +229,42 @@ def larmor_freq(self, B0=9.4):
         """
         return -self.gyromagnetic_ratio * B0
 
+    def ref_freq_to_B0(self, ref_freq=400):
+        """Return the magnetic field strength B0 given the primary reference frequency.
+
+        Args:
+            float ref_freq: primary reference frequency in MHz
+
+        Returns:
+            float: magnetic flux density in T
+
+        Example
+        -------
+
+        >>> H1 = Isotope(symbol="1H")
+        >>> B0 = H1.ref_freq_to_B0(ref_freq = 400)
+        """
+        isotope_data = isotope_reference(self.symbol)
+        return 0.02348731439404777 * ref_freq / isotope_data.ratio
```

Comment:
When you add the reference property from last comment this code can be reduced to:

```py
return 0.02348731439404777 * ref_freq / self.reference.ratio
```